### DISCUSSION TITLE
Add decimal option to round()

### DISCUSSION THREAD

**Discussion by alexis-bvq** [atoti/atoti#626] [General]
## Description
The round measure exists but only for integers.
If we could pass in argument the number of digits we want to round to:
tt.round(measure, digits=int)
Digits would be the power of 10 you want to round at for example.

For simcorp we need to round at 10**(-2) : 
1000.23556 => 1000.24
1000.234 => 1000.23
1000.2 => 1000.2

cheers

**Comment by fabiencelier** [comment]
A workaround is to multiply it by a factor, round it then divide by the factor:

```python
import math

def round_with_digits(measure, digits: int):
   return tt.round(measure * 10**digits) / 10**digits
```

But I agree that we could implement it in a clean efficient way in the core atoti

We probably want to keep the casting to long when rounding to an integer, so we could take digits as a int (not a measure)  and cast to long if it is 0 and cast to double otherwise.

Also[ math.floor](https://docs.python.org/3/library/math.html#math.floor) and [math.ceil](https://docs.python.org/3/library/math.html#math.ceil) in Python does not have the number of digits while [round()](https://docs.python.org/3/library/functions.html#round) has it, so I'm not sure if it should be done for ceil and floor.

**Comment by rcolle** [comment]
In Python:
> rounding is done toward the even choice (so, for example, both round(0.5) and round(-0.5) are 0, and round(1.5) is 2).

That's not what we do in Java (`Math.round` rounds ties towards positive infinity). It's probably worth looking into following Python's specification, since this method has less bias.

**Comment by tibdex** [comment]
We need to benchmark code such as:

```java
BigDecimal.valueOf(number).setScale(decimals, BigDecimal. HALF_EVEN).floatValue()
```
